server/session/player.go

```diff
@@ -553,15 +570,79 @@ func stackToItem(it protocol.ItemStack) item.Stack {
 	return nbtconv.ReadItem(it.NBTData, &s)
 }
 
+// itemToRecipeIngredientItem converts a recipe.Item into a type that can be used over the protocol.
+func itemToRecipeIngredientItem(s recipe.Item) protocol.RecipeIngredientItem {
+	if s.Item() == nil {
+		return protocol.RecipeIngredientItem{}
+	}
+	rid, meta, ok := world.ItemRuntimeID(s.Item())
+	if !ok {
+		panic("should never happen")
+	}
+
+	if s.AllTypes {
+		meta = craftingFlagAll
+	}
+
+	return protocol.RecipeIngredientItem{
+		NetworkID:     rid,
+		MetadataValue: int32(meta),
+		Count:         int32(s.Count()),
+	}
+}
+
+// itemsToRecipeIngredientItems converts a list of recipe.Items into a type that can be used over the protocol.
+func itemsToRecipeIngredientItems(s []recipe.Item) (r []protocol.RecipeIngredientItem) {
+	for _, st := range s {
+		r = append(r, itemToRecipeIngredientItem(st))
+	}
+	return
+}
+
+// stripDamageFromProtocolItem strips the damage from a protocol item.
+func stripDamageFromProtocolItem(st protocol.ItemStack) protocol.ItemStack {
```

Comment:
Might want to just name this `deleteDamage` or something like it, the function signature makes the latter part clear enough.